### ISSUE TITLE
Unexpected merge of # NOQA flags

### ISSUE THREAD

**Issue by SofienBenAyed** [timothycrosley/isort#1233]
I have some `# NOQA` flags in my imports and something that seems unexpected happened with the following imports:
```python
from project.moduleA.moduleB.moduleC.moduleD.moduleE.moduleF import MyClassA  # NOQA
from project.moduleA.moduleB.moduleC.moduleD.moduleE.moduleF import (  # NOQA
    MyVeryUsefulClassB,
)
```

When running `isort`, the second import is devoid of its `# NOQA` and the first one gets both.
```python
from project.moduleA.moduleB.moduleC.moduleD.moduleE.moduleF import (
    MyClassA  # NOQA; NOQA,
)
from project.moduleA.moduleB.moduleC.moduleD.moduleE.moduleF import MyVeryUsefulClassB
```

Moreover, it makes it incompatible with `black`.

Thank you!

---

Below is the `isort v4.3.21` configuration I use for `black` compatibility.
```
[isort]
multi_line_output=3
include_trailing_comma=True
force_grid_wrap=0
use_parentheses=True
line_length=88
```

**Comment by timothycrosley** [comment]
Thanks for reporting! This should be fixed on isort 5.0.0: 
https://timothycrosley.github.io/isort/

**Comment by SofienBenAyed** [comment]
Hi @timothycrosley! I've tried again on the same example using `isort v5.4.2` and I encounter the same issue.
The output is:
```python
from project.moduleA.moduleB.moduleC.moduleD.moduleE.moduleF import (
    MyClassA  # NOQA; NOQA
)
from project.moduleA.moduleB.moduleC.moduleD.moduleE.moduleF import MyVeryUsefulClassB
```
